Gosu/Audio.hpp

```diff
@@ -80,59 +90,7 @@ namespace Gosu
         //! or low values. Use 1.0 for normal playback speed.
         Channel play_pan(double pan, double volume = 1, double speed = 1,
             bool looping = false) const;
-    };
-
-    //! Songs are less flexible than samples. Only Song can be played at any given time,
-    //! and there is no way to control its pan (stereo position) or speed.
-    class Song
-    {
-        class BaseData;
-        class ModuleData;
-        class StreamData;
-        std::unique_ptr<BaseData> data;
-        
-        // Non-movable to avoid dangling internal references.
-        Song(Song&&) = delete;
-        // Non-movable to avoid dangling internal references.
-        Song& operator=(Song&&) = delete;
 
-    public:
-        //! Constructs a song that can be played on the provided audio system
-        //! and loads the song from a file. The type is determined from the
-        //! filename.
-        explicit Song(const std::string& filename);
-        
-        //! Constructs a song of the specified type that can be played on the
-        //! provided audio system and loads the song data from a stream.
-        explicit Song(Reader reader);
-        
-        ~Song();
-        
-        //! Returns the song currently being played or paused, or 0 if
-        //! no song has been played yet or the last song has finished
-        //! playing.
-        static Song* current_song();
-        
-        //! Starts or resumes playback of the song. This will stop all other
-        //! songs and set the current song to this object.
-        void play(bool looping = false);
-        //! Pauses playback of the song. It is not considered being played.
-        //! current_song will stay the same.
-        void pause();
-        //! Returns true if the song is the current song, but in paused
-        //! mode.
-        bool paused() const;
-        //! Stops playback of this song if it is currently played or paused.
-        //! Afterwards, current_song will return nullptr.
-        void stop();
-        //! Returns true if the song is currently playing.
-        bool playing() const;
-        //! Returns the current volume of the song.
-        double volume() const;
-        //! \param volume Can be anything from 0.0 (silence) to 1.0 (full
-        //! volume).
-        void set_volume(double volume);
-        
         //! Called every tick by Window for management purposes.
         static void update();
```

Comment:
Is there a good reason why this can't be available in Ruby as well? It would at least to some degree solve issue #390 and has no negative side effects as far as I can tell. This way we can get rid of the window in the tests and the guy who used gosu just to play music needs some kind of mainloop one way or the other. I'm not sure if playing sounds without window should work out of the box in Gosu, as its a "game development" lib :smile:

Reply:
Good point. Maybe there's some way in which Gosu could spawn a thread that keeps all sounds running, but in the meantime, Ruby users should have access to this.